inst/tests/tests.Rraw

```diff
@@ -6207,7 +6208,11 @@ test(1464.13, rleidv(DT, 1:2), ans<-INT(1,2,3,4,5,6,6,6,7,8,8,9,10,11,12,13,14,1
 test(1464.14, rleidv(DT, 2:1), ans)
 test(1464.15, rleidv(DT, c(3,1)), INT(1,1,2,2,3,4,5,5,6,7,8,9,10,11,12,13,14,15,16,17))
 
-if (test_xts) {
+if (test_xts && 
+     # package xts also has an issue with a && clause.  To avoid the tests failing due to this 
+     # package we only test xts when not checking OR if the package version has incremented
+     # (either because my diagnosis was incorrect or the bug was fixed)
+    {!nzchar(Sys.getenv("_R_CHECK_LENGTH_1_LOGIC2_")) || packageVersion("xts") > "0.11-1"}) {
```

Comment:
is this code, precisely `!nzchar` handle properly the case when `_R_CHECK_LENGTH_1_LOGIC2_` will be set to `false`?

Reply:
Yes because false doesn't turn it off. false means warning not error. From R's news:
> Experimentally, setting environment variable ‘_R_CHECK_LENGTH_1_LOGIC2_’ will lead to warnings (or errors if the variable is set to a ‘true’ value) when ‘&&’ or ‘||’ encounter and use arguments of length more than one.